Lib/python/pyrun.swg

```diff
@@ -127,7 +127,11 @@ SWIG_Python_AppendOutput(PyObject* result, PyObject* obj) {
     if (!PyList_Check(result)) {
       PyObject *o2 = result;
       result = PyList_New(1);
-      PyList_SetItem(result, 0, o2);
+      if (result) {
+        PyList_SET_ITEM(result, 0, o2);
+      } else {
+        return o2;
```

Comment:
This is another OOM fix - but as well as giving a sane return value when we fail to append (i.e. the object as-was), we should also call `PyErr_NoMemory()` , so the caller is made aware that the append failed. I'll just push a change to do that...

Reply:
Scratch that: `PyList_New` \(and others\) all call through to `_PyObject_GC_Malloc` which has this:
```c
    g = (PyGC_Head *)PyObject_MALLOC(
        sizeof(PyGC_Head) + basicsize);
    if (g == NULL)
        return PyErr_NoMemory();
```
so the exception will already be set up when we get `NULL` back...